pkg/cmd/inspect.go

```diff
@@ -0,0 +1,184 @@
+	if len(args) == 0 {
+		return errors.New("no integration files have been provided, nothing to inspect")
+	}
+
+	// Ensure source files exist.
+	for _, arg := range args {
+		// fmt.Printf("Validating file: %v\n", arg)
+		fileExists, err := util.FileExists(arg)
+
+		// Report any error.
+		if err != nil {
+			return err
+		}
+
+		// Signal file not found.
+		if !fileExists {
+			return errors.New("input file " + arg + " file does not exist")
+		}
+	}
+
+	// If provided, ensure that that the dependencies directory exists.
+	if command.DependenciesDirectory != "" {
+		dependenciesDirectoryExists, err := util.DirectoryExists(command.DependenciesDirectory)
+		// Report any error.
+		if err != nil {
+			return err
+		}
+
+		// Signal file not found.
+		if !dependenciesDirectoryExists {
+			return errors.New("input file " + command.DependenciesDirectory + " file does not exist")
+		}
+	}
+
+	// TODO: validate list of additional dependencies i.e. make sure that each dependency is
+	// in the correct format.
+
+	return nil
+}
+
+func (command *inspectCmdOptions) run(args []string) error {
+	// A Camel catalog is requiref for this operatio.
+	settings := ""
+	mvn := v1.MavenSpec{
+		LocalRepository: "",
+	}
+	runtime := v1.RuntimeSpec{
+		Version:  "1.3.0",
+		Provider: v1.RuntimeProviderMain,
+	}
+	providerDependencies := []maven.Dependency{}
+	catalog, err := camel.GenerateLocalCatalog(settings, mvn, runtime, providerDependencies)
+	if err != nil {
+		return err
+	}
+
+	// TODO: compression not supported for this command for now.
+	compression := false
+
+	// List of top-level dependencies.
+	dependencies := strset.New()
+
+	// Invoke the dependency inspector code for each source file.
+	for _, source := range args {
+		data, _, err := loadData(source, compression, compression)
+		if err != nil {
+			return err
+		}
+
+		sourceSpec := v1.SourceSpec{
+			DataSpec: v1.DataSpec{
+				Name:        path.Base(source),
+				Content:     data,
+				Compression: compression,
+			},
+		}
+
+		// Extract list of top-level dependencies.
+		metadata.Extract(catalog, sourceSpec)
+		meta := metadata.Extract(catalog, sourceSpec)
+		lang := sourceSpec.InferLanguage()
+
+		// add auto-detected dependencies
+		dependencies.Merge(meta.Dependencies)
+
+		for loader, v := range catalog.Loaders {
+			// add loader specific dependencies
+			if sourceSpec.Loader != "" && sourceSpec.Loader == loader {
+				dependencies.Add(fmt.Sprintf("mvn:%s/%s", v.GroupID, v.ArtifactID))
+
+				for _, d := range v.Dependencies {
+					dependencies.Add(fmt.Sprintf("mvn:%s/%s", d.GroupID, d.ArtifactID))
+				}
+			} else if sourceSpec.Loader == "" {
+				// add language specific dependencies
+				if util.StringSliceExists(v.Languages, string(lang)) {
+					dependencies.Add(fmt.Sprintf("mvn:%s/%s", v.GroupID, v.ArtifactID))
+
+					for _, d := range v.Dependencies {
+						dependencies.Add(fmt.Sprintf("mvn:%s/%s", d.GroupID, d.ArtifactID))
+					}
+				}
+			}
+		}
+	}
```

Comment:
wonder if this block could be extricated in an utility method so so it won't break in case the logic changes